control-plane/csi-controller/src/main.rs

```diff
@@ -97,10 +97,23 @@ pub async fn main() -> Result<(), String> {
         CsiControllerConfig::get_config().rest_endpoint()
     );
 
-    server::CsiServer::run(
-        args.value_of("socket")
-            .expect("CSI socket must be specfied")
-            .to_string(),
-    )
-    .await
+    let mut signal_term = tokio::signal::unix::signal(tokio::signal::unix::SignalKind::terminate())
+        .expect("Failed to register handler for SIGTERM");
+    let mut signal_int = tokio::signal::unix::signal(tokio::signal::unix::SignalKind::interrupt())
+        .expect("Failed to register handler for SIGINT");
+
+    tokio::select! {
+        _srv =  server::CsiServer::run(args.value_of("socket")
+            .expect("CSI socket must be specified")
+            .to_string()) => {
+            _srv?
+        }
+        _evt = signal_term.recv() => {
+                opentelemetry::global::force_flush_tracer_provider();
+            }
+        _evt = signal_int.recv() => {
+                opentelemetry::global::force_flush_tracer_provider();
+            }
+    }
```

Comment:
really?

Reply:
cryptic comment, I will assume that we shouldn't invoke `opentelemetry::global::force_flush_tracer_provider()`
Fixed.
https://github.com/mayadata-io/mayastor-control-plane/pull/158/commits/fd8a717b34e9e4bc0a8d546f8732f47f0a1d5edc

Reply:
I think we should keep it actually, it would be useful to flush the traces.